irohad/torii/impl/command_service.cpp

```diff
@@ -130,4 +136,117 @@ namespace torii {
     Status(*request, *response);
     return grpc::Status::OK;
   }
+
+  void CommandService::StatusStream(
+      iroha::protocol::TxStatusRequest const &request,
+      grpc::ServerWriter<iroha::protocol::ToriiResponse> &response_writer) {
+    auto resp = cache_->findItem(request.tx_hash());
+    if (resp) {
+      if (resp->tx_status()
+              == iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED
+          or resp->tx_status()
+              == iroha::protocol::TxStatus::STATEFUL_VALIDATION_FAILED
+          or resp->tx_status() == iroha::protocol::TxStatus::NOT_RECEIVED
+          or resp->tx_status() == iroha::protocol::TxStatus::COMMITTED) {
+        response_writer.WriteLast(*resp, grpc::WriteOptions());
+        return;
+      }
+      response_writer.Write(*resp);
+    }
+
+    bool finished = false;
+    auto subscription = rxcpp::composite_subscription();
+    auto request_hash = request.tx_hash();
+
+    tx_processor_->transactionNotifier()
+        .filter([&request_hash](auto response) {
+          return response->tx_hash == request_hash;
+        })
+        .subscribe(
+            subscription,
+            [&](std::shared_ptr<iroha::model::TransactionResponse>
+                    iroha_response) {
+              iroha::protocol::ToriiResponse resp_sub;
+              resp_sub.set_tx_hash(request_hash);
+
+              switch (iroha_response->current_status) {
+                case iroha::model::TransactionResponse::
+                    STATELESS_VALIDATION_FAILED:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED);
+                  response_writer.WriteLast(resp_sub, grpc::WriteOptions());
+                  subscription.unsubscribe();
+                  finished = true;
```

Comment:
The code inside case statements looks duplicated. Maybe better to make a template functions for this stuff?